src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientEventService.kt

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.gateways.PrisonApiGateway
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.entities.RestrictedPatient
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.response.Agency
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.response.MovementResponse
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.repositories.RestrictedPatientsRepository
+
+@Service
+class RestrictedPatientEventService(
+  private val prisonApiApplicationGateway: PrisonApiGateway,
+  private val domainEventPublisher: DomainEventPublisher,
+  private val restrictedPatientsRepository: RestrictedPatientsRepository,
+) {
```

Comment:
So the majority of this change is moving the event handling our of RestrictedPatientService into its own class with the application scope prison api client